src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsActions.ts

```diff
@@ -0,0 +1,253 @@
+import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
+import { IKeybindingService } from 'vs/platform/keybinding/common/keybinding';
+import { ILogService } from 'vs/platform/log/common/log';
+import { getErrorMessage } from 'vs/base/common/errors';
+import * as types from 'vs/base/common/types';
+import { ActionBar, ActionsOrientation } from 'vs/base/browser/ui/actionbar/actionbar';
+import { Separator } from 'sql/base/browser/ui/separator/separator';
+import { ToggleMoreActions } from 'sql/workbench/contrib/notebook/browser/cellToolbarActions';
+import { NotebookViewsExtension } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViewsExtension';
+import { INotebookView } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViews';
+
+export class ViewSettingsAction extends Action {
+	private static readonly ID = 'viewSettings';
+	private static readonly LABEL = undefined;
+	private static readonly ICON = 'notebook-button settings masked-icon';
+
+	constructor(
+		private _context: NotebookViewsExtension,
+		@IInstantiationService private _instantiationService: IInstantiationService,
+	) {
+		super(ViewSettingsAction.ID, ViewSettingsAction.LABEL, ViewSettingsAction.ICON);
+	}
+
+	override run(): Promise<void> {
+		return new Promise<void>((resolve) => {
+			const optionsModal = this._instantiationService.createInstance(ViewOptionsModal, this._context.getActiveView());
+			optionsModal.render();
+			optionsModal.open();
+		});
+	}
+}
+
+export class DeleteViewAction extends Action {
+	private static readonly ID = 'viewSettings';
+	private static readonly LABEL = undefined;
+	private static readonly ICON = 'notebook-button delete masked-icon';
+
+	constructor(
+		private _notebookViews: NotebookViewsExtension,
+		@IDialogService private readonly dialogService: IDialogService,
+		@INotificationService private readonly notificationService: INotificationService
+	) {
+		super(DeleteViewAction.ID, DeleteViewAction.LABEL, DeleteViewAction.ICON);
+	}
+
+	override async run(): Promise<void> {
+		const activeView = this._notebookViews.getActiveView();
+		if (activeView) {
+			const confirmDelete = await this.confirmDelete(activeView);
+			if (confirmDelete) {
+				this._notebookViews.removeView(activeView.guid);
+				this._notebookViews.notebook.viewMode = ViewMode.Notebook;
+			}
+		} else {
+			this.notificationService.error(localize('viewsUnableToRemove', "Unable to remove view"));
+		}
+	}
+
+	private async confirmDelete(view: INotebookView): Promise<boolean> {
+		const result = await this.dialogService.confirm({
+			message: localize('confirmDelete', "Are you sure you want to delete view \"{0}\"?", view.name),
+			primaryButton: localize('delete', "&&Delete"),
+			type: 'question'
+		});
+
+		if (result.confirmed) {
+			return true;
+		}
+		return false;
+	}
+}
+
+export class InsertCellAction extends Action {
+	private static readonly ID = 'viewSettings';
```

Comment:
Is this ID correct? It should be insertCell or something right?

And generally we should prefix action IDs with an area, e.g. `notebookView.viewSettings`

Reply:
Thanks for catching this. I corrected the ID and appended a prefix to the rest of the actions on this file.